### DISCUSSION TITLE
Are you open to PRs for to make either of these configurable?

### DISCUSSION THREAD

**Discussion by michael-odell** [marlonrichert/zsh-autocomplete#520] [General]
Thanks for making this nifty tool and sharing it with the world.  I specifically like the idea of having completions being generated and displayed for me as I'm typing.  Even in just a day I've seen completions I didn't realize I had set up.  And it's a great way to get reminders about options and their meanings, even for some tools that don't have manpages.

I ran into a couple of things that didn't work as I expected or as I'm in the habit of working.  I wanted to reach out and see if you're open to PRs around them and if so, to gather information about how you'd like to see them implemented.

### Disabling incremental history functions

I have years of muscle memory attached to the way that the incremental history search works in zsh.  I frequently hit Ctrl-R, type a few letters I think I remember from a recent command, and then Ctrl-R, Ctrl-R, Ctrl-R as I keep hopping back through history entries until I find the correct one that I wanted to run.  Next I'll sometimes Ctrl-A to go to the beginning of the line and modify the command before pushing enter to execute it.

For me, I'd prefer work with out-of-the-box zsh incremental history than what's here.  I don't discount the use of what you've done for others, but I wanted to the ability to turn it off.

### Difficult to read colors in unambiguous mode

I'm using the "unambiguous" mode for autocompletions (i.e. `zstyle ':autocomplete:*' insert-unambiguous yes
`, and in that mode there appears to be a hardcoded color that I find difficult to read in my terminal.

Here's a screenshot of an ambiguous completion for me.  I can't quickly read the dark text on dark background (each of which contains `pod` here).  It's possible my terminal has unusual configuration to make the colors look like that -- I'm using iTerm 2 with a light solarized color scheme.

<img width="779" alt="image" src="https://user-images.githubusercontent.com/844498/205416750-edde462c-6534-4a99-b6e5-91d0855cdb80.png">

### Can I help?

I hacked through the above issues in probably careless fashion today so I could try out the other features of this plugin.  If you're interested, you can see what I did here:

1. Disabling history-related code: https://github.com/michael-odell/zsh-autocomplete/commit/ff8f2760732e22d8e9f7d53b8bae6f681bf93355
2. Removing the unreadable colors: https://github.com/michael-odell/zsh-autocomplete/commit/c67b47b1170f088f886aeb963e77bf379e306c1b

Again, these were hacks.  If you're interested in adding one or both to your repo, I'd be glad to tidy them up and submit PRs.

I'm not an expert at configuring zsh, but I'm a decent bash/posix shell scripter.  If you think either of these is valuable, please guide me a little in how you think I should expose the configuration (i.e. should the user set an environment variable in .zshrc?  zstyle?  something else?) and I'll give it a shot.

Even if you're not interested in PRs, I'd be grateful for any advice you have about implementing these changes the right way rather than the grep-and-comment-or-delete workflow I used so far.

**Comment by marlonrichert** [comment]
To revert to the default functionality of `^R`, add this to your `.zshrc` file:

```zsh
zle -N {,.}history-incremental-search-backward
```

I’d be happy to accept a pull request for documenting this in the [example `.zshrc` file](https://github.com/marlonrichert/zsh-autocomplete/blob/main/.zshrc).

It looks like your terminal’s color scheme renders bright yellow as gray, which is nonstandard, to say the least.

I’d be open to making this highlighting configurable, though. You’d have to add a new `zstyle` for this over here:

https://github.com/marlonrichert/zsh-autocomplete/blob/866c6610bfeea18b055001703822c9ffd92337f7/functions/completion/_autocomplete.unambiguous#L27

Overall, looking at your code, I think you made more changes than is necessary. When writing a pull request, try to adhere to the following principles:

* [Don’t Repeat Yourself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
* [Keep It Simple Stupid](https://en.wikipedia.org/wiki/KISS_principle)
* [You Aren’t Gonna Need It](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

**Comment by michael-odell** [comment]
Thanks for all this.  I didn't know quite how to respond and am still finding my way through troubleshooting a bit.  As is probably true for you, I don't spend _all_ my time on my shell configuration.   I just fix things when I get a few moments.

I'm not sure I quite this suggestion enough to offer up a PR for documentation.  The way things seemed to me when I used the following block in my zshrc was that _some_ of the history-affecting functionality may have been disabled but things were still not quite the same as if I didn't load your plugin at all.  For instance, I found `^R` to do the thing I expected, but also clear any text to the left of the cursor.

```
zle -N {,.}history-incremental-search-backward
``` 

It's entirely possible this was my own mistake, but I don't feel comfortable writing documentation to suggest how to do something without understanding it.

As far as the configurable highlight color, I may still send you a PR -- but that'll depend on my getting everything working with the above and my focus isn't on it at present.

Thank you for the prompt response -- I apologize for not returning the favor.  I'll follow back up here if I can more clearly identify the problem with my configuration and the `zle -N` suggestion or if I do indeed implement a configurable highlight color as above.
